tests/test_ccdoffsets.py

```diff
@@ -0,0 +1,102 @@
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+"""Test the fgcmcal computeCcdOffsets code with testdata_jointcal.
+
+"""
+
+import unittest
+import os
+
+import lsst.utils
+import lsst.daf.persistence as dafPersist
+
+import lsst.fgcmcal as fgcmcal
+
+
+class FgcmCcdOffsetsTest(lsst.utils.tests.TestCase):
+    @classmethod
+    def setUpClass(cls):
+        try:
+            cls.dataDir = lsst.utils.getPackageDir('testdata_jointcal')
+        except lsst.pex.exceptions.NotFoundError:
+            raise unittest.SkipTest("testdata_jointcal not setup")
+
+    def test_fgcmCcdOffsetsHsc(self):
+        """
+        Test computation of ccd offsets for HSC.
+        """
+
+        lsst.log.setLevel("HscMapper", lsst.log.FATAL)
+
+        butler = dafPersist.Butler(os.path.join(self.dataDir, 'hsc'))
+
+        visit = 903986
+        ccd = 16
+
+        visitInfo = butler.get('calexp_visitInfo', visit=visit, ccd=ccd)
+        rotAngle = visitInfo.getBoresightRotAngle().asDegrees()
+
+        camera = butler.get('camera')
+
+        ccdOffsets = fgcmcal.utilities.computeCcdOffsets(camera, rotAngle)
+
+        # Spot check relative orientations of some of the ccds
+        # This is based on
+        # https://subarutelescope.org/Observing/Instruments/HSC/CCDPosition_20170212.png
+        self.assertLess(ccdOffsets['DELTA_RA'][15], ccdOffsets['DELTA_RA'][10])
+        self.assertLess(ccdOffsets['DELTA_RA'][95], ccdOffsets['DELTA_RA'][90])
+        self.assertGreater(ccdOffsets['DELTA_RA'][46], 0.0)
+
+        self.assertLess(ccdOffsets['DELTA_DEC'][15], ccdOffsets['DELTA_DEC'][95])
+        self.assertLess(ccdOffsets['DELTA_DEC'][10], ccdOffsets['DELTA_DEC'][90])
+        self.assertGreater(ccdOffsets['DELTA_DEC'][97], 0.0)
+
+        # Check the sizes
+        rotatedCcds = [100, 101, 102, 103]
+
+        for i, ccd in enumerate(ccdOffsets['CCDNUM']):
+            if ccd in rotatedCcds:
+                self.assertLess(ccdOffsets['RA_SIZE'][i], ccdOffsets['DEC_SIZE'][i])
+                self.assertGreater(ccdOffsets['DEC_SIZE'][i], 0.17)
+                self.assertLess(ccdOffsets['DEC_SIZE'][i], 0.20)
+                self.assertGreater(ccdOffsets['RA_SIZE'][i], 0.07)
+                self.assertLess(ccdOffsets['RA_SIZE'][i], 0.10)
```

Comment:
Where did these sizes come from, and what are the units here? Why the `0.03` range in each case? Please document that in the code.